// @clevercloud/client/cjs/oauth.node.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.addOauthHeader = addOauthHeader;

var _oauth = _interopRequireDefault(require("oauth-1.0a"));

var _crypto = _interopRequireDefault(require("crypto"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function addOauthHeader(tokens) {
  return function (requestParams) {
    const {
      method,
      url,
      headers,
      queryParams
    } = requestParams;
    const oauth = (0, _oauth.default)({
      consumer: {
        key: tokens.OAUTH_CONSUMER_KEY,
        secret: tokens.OAUTH_CONSUMER_SECRET
      },
      signature_method: 'HMAC-SHA512',

      hash_function(baseString, key) {
        return _crypto.default.createHmac('sha512', key).update(baseString).digest('base64');
      }

    });
    const requestData = {
      url,
      method,
      data: queryParams
    };
    const oauthData = oauth.authorize(requestData, {
      key: tokens.API_OAUTH_TOKEN,
      secret: tokens.API_OAUTH_TOKEN_SECRET
    });
    const oauthHeaders = oauth.toHeader(oauthData);
    return { ...requestParams,
      headers: { ...oauthHeaders,
        ...headers
      }
    };
  };
}
